open-company/open-company-web
# Slack notification step

Card: https://trello.com/c/LxKMCjvB

Related PR:
- Auth https://github.com/open-company/open-company-auth/pull/46

To prepare tests:
- clean up your local DBs or at least remove any team/org/slack-org associated with our 2 test slack orgs
- remove the Carrot local app from our 2 test slack orgs (from https://oc-test-org1.slack.com/apps/A5SGHA79P-carrot-local-development?page=1 and https://oc-test-org2.slack.com/apps/A5SGHA79P-carrot-local-development?page=1)

Test 1:
- logout from localhost
- signup with Slack test org 1 (as the first owner user is ok)
- [x] you went through the first Slack auth step? Good
- [x] you did NOT go through the second Slack auth step? Good
- set profile data
- set org data
- land on org digest
- [x] do you see the add Slack bot banner? Good
- click Learn More
- [x] do you see the modal? Good
- click add at the bottom
- [x] do you see the Slack second step page? Good
- click back
- now click Add directly from the banner
- [x] do you see the Slack second step again? Good
- grant permissions
- [x] are you back to your digest? Good
- [x] no settings panel open? Good
- [x] did you get the Slack bot successfully added notification? Good

Test 2:
- logout from localhost
- signup with email
- setup profile
- setup team/org
- land on digest
- [x] do you NOT see the add Slack bot banner? Good
- go to Digest Settings
- add Slack team
- add the Test Org 2
- [x] you went back to org digest with Digest Settings open? Good
- [x] do you get the add Slack bot banner? Good
- add the slack bot from here or the learn more modal
- grant second set of permissions
- [x] are you back to your org digest with Digest Settings open? Good
- [x] did you get the bot added notification? Good
- [x] check that you can load the roster the channels etc (invite a slack user for example or share to a channel)
